llvm/lib/Transforms/Vectorize/VPlanAnalysis.cpp

```diff
@@ -50,6 +50,12 @@ Type *VPTypeAnalysis::inferScalarTypeForRecipe(const VPInstruction *R) {
     return SetResultTyFromOp();
 
   switch (Opcode) {
+  case Instruction::ExtractElement: {
+    Type *ResTy = inferScalarType(R->getOperand(0));
+    VPValue *OtherV = R->getOperand(1);
+    CachedTypes[OtherV] = ResTy;
```

Comment:
Can the index type not be different from the result element type? Does it still work if we just remove the CachedTypes[OtherV] bit

Reply:
Ah perhaps I've misunderstood what `CachedTypes[OtherV]` was doing. I was basically just copying what the other code did without really understanding it, but now it looks wrong. I'll delete the caching.